modules/sr/robot/robot.py

```diff
@@ -134,7 +134,8 @@ def _init_ruggeduinos(self) -> None:
         self.ruggeduinos = ruggeduino.init_ruggeduino_array(self.webot)
 
     def _init_camera(self) -> None:
-        self.camera = camera.Camera(self.webot)
+        # See comment in Camera.see for why we need to pass the step lock here.
+        self.camera = camera.Camera(self.webot, self._step_lock)
```

Comment:
I think I'd actually prefer that we didn't pass the lock downwards and instead facaded the `see` method. However since `self.camera` is implicitly part of the API here I think we need to do this this way for now.

I suggest we revisit this at some point and make `self.camera` private (i.e: `self._camera`) and then move the locking back up into an explicit `Robot.see` method.